dev/Deployment/DeploymentManager.cpp

```diff
@@ -0,0 +1,249 @@
+        std::wstring frameworkFamilyName{ frameworkPackageInfo.Package(0).packageFamilyName };
+        auto posFrameworkIdentifier{ frameworkFamilyName.find(WINDOWSAPPRUNTIME_FRAMEWORK_PACKAGE_IDENTIFIER) };
+        const int c_frameworkIdentifierLength{ ARRAYSIZE(WINDOWSAPPRUNTIME_FRAMEWORK_PACKAGE_IDENTIFIER) - 1 };
+        auto posFrameworkDelimeter{ frameworkFamilyName.find(WINDOWSAPPRUNTIME_NAME_DELIMETER) };
+        auto packageFamilyPrefix{ frameworkFamilyName.substr(0, posFrameworkIdentifier)};
+        auto packageFamilyPostfix{ frameworkFamilyName.substr(posFrameworkIdentifier + c_frameworkIdentifierLength, posFrameworkDelimeter - (posFrameworkIdentifier + c_frameworkIdentifierLength))};
+
+        // Loop through all of the target packages and validate.
+        HRESULT verifyResult{ S_OK };
+        for (const auto& package : c_targetPackages)
+        {
+            // Build package family name based on the framework naming scheme.
+            std::wstring packageFamilyName{ packageFamilyPrefix + package.identifier + packageFamilyPostfix + WINDOWSAPPRUNTIME_NAME_SUFFIX };
+
+            // Get target version based on the framework.
+            auto targetPackageVersion{ frameworkPackageInfo.Package(0).packageId.version };
+
+            verifyResult = VerifyPackage(packageFamilyName, targetPackageVersion);
+            if (FAILED(verifyResult))
+            {
+                break;
+            }
+        }
+
+        auto status{ winrt::make<implementation::DeploymentStatus>(SUCCEEDED(verifyResult), FAILED(verifyResult), verifyResult) };
+        return status;
+    }
+
+    winrt::Microsoft::Windows::ApplicationModel::WindowsAppSDK::DeploymentStatus DeploymentManager::Initialize()
+    {
+        THROW_HR_IF(E_NOTIMPL, !::Microsoft::Windows::ApplicationModel::WindowsAppSDK::Feature_DeploymentAPI::IsEnabled());
+        FAIL_FAST_HR_IF(HRESULT_FROM_WIN32(APPMODEL_ERROR_NO_PACKAGE), !AppModel::Identity::IsPackagedProcess());
+        return Initialize(GetCurrentFrameworkPackageFullName());
+    }
+
+    winrt::Microsoft::Windows::ApplicationModel::WindowsAppSDK::DeploymentStatus DeploymentManager::Initialize(hstring const& packageFullName)
+    {
+        THROW_HR_IF(E_NOTIMPL, !::Microsoft::Windows::ApplicationModel::WindowsAppSDK::Feature_DeploymentAPI::IsEnabled());
+        auto status{ DeploymentManager::GetStatus(packageFullName) };
+        if (status.IsOK())
+        {
+            return status;
+        }
+
+        std::wstring frameworkPackageFullName{ packageFullName };
+        auto initializeResult{ DeployPackages(frameworkPackageFullName) };
+        auto initializeStatus{ winrt::make<implementation::DeploymentStatus>(SUCCEEDED(initializeResult), FAILED(initializeResult), initializeResult) };
+        return initializeStatus;
+    }
+
+    MddCore::PackageInfo DeploymentManager::GetPackageInfoForPackage(std::wstring const& packageFullName)
```

Comment:
`MddCore::*` weren't expected to be an API (even an internal one). But that's cool. I'll work on polishing/liberating these things in the future so it's more generally useful (cleanly, w/o unexpected future surprises :-)